client/driver/docker.go

```diff
@@ -1458,10 +1463,48 @@ func calculatePercent(newSample, oldSample, newTotal, oldTotal uint64, cores int
 	return (float64(numerator) / float64(denom)) * float64(cores) * 100.0
 }
 
+func authOptionFromDriverConfig(auth DockerDriverAuth) (*docker.AuthConfiguration, error) {
+	return &docker.AuthConfiguration{
+		Username:      auth.Username,
+		Password:      auth.Password,
+		Email:         auth.Email,
+		ServerAddress: auth.ServerAddress,
+	}, nil
+}
+
+func authOptionFromHelper(helperName, repoName string, clientConfig *config.Config) (*docker.AuthConfiguration, error) {
+	helper := dockerAuthHelperPrefix + helperName
+	cmd := exec.Command(helper, "get")
+	cmd.Stdin = strings.NewReader(repoName)
+
+	// For helper to be able to read credentials from vault
+	cmd.Env = append(os.Environ(), fmt.Sprintf("VAULT_TOKEN=%s", clientConfig.VaultConfig.Token))
```

Comment:
I would like to see an operator option to allow passing the vault token to credential helpers. I would probably treat it as a white list of allowed helpers. You generally do not want to give that out any more than you have to! Isn't security fun :)